tests/Integration/Behaviour/Features/Context/Domain/ApiAccessManagementFeatureContext.php

```diff
@@ -0,0 +1,303 @@
+    {
+        ApiAccessResetter::resetApiAccess();
+    }
+
+    private function fixDataType(array $data): array
+    {
+        if (array_key_exists('enabled', $data) && !is_null($data['enabled'])) {
+            $data['enabled'] = filter_var($data['enabled'], FILTER_VALIDATE_BOOL);
+        }
+
+        return $data;
+    }
+
+    private function getUnicityConstraintErrorCode(string $fieldName): int
+    {
+        $constraintErrorFieldMap = [
+            'clientId' => ApiAccessConstraintException::CLIENT_ID_ALREADY_USED,
+            'clientName' => ApiAccessConstraintException::CLIENT_NAME_ALREADY_USED,
+        ];
+
+        if (!array_key_exists($fieldName, $constraintErrorFieldMap)) {
+            throw new \RuntimeException(sprintf('"%s" is not mapped with constraint error code', $fieldName));
+        }
+
+        return $constraintErrorFieldMap[$fieldName];
+    }
+
+    private function getInvalidConstraintErrorCode(string $fieldName): int
+    {
+        $constraintErrorFieldMap = [
+            'apiClientId' => ApiAccessConstraintException::INVALID_CLIENT_ID,
+            'clientName' => ApiAccessConstraintException::INVALID_CLIENT_NAME,
+            'enabled' => ApiAccessConstraintException::INVALID_ENABLED,
+            'description' => ApiAccessConstraintException::INVALID_DESCRIPTION,
+        ];
+
+        if (!array_key_exists($fieldName, $constraintErrorFieldMap)) {
+            throw new \RuntimeException(sprintf('"%s" is not mapped with constraint error code', $fieldName));
+        }
+
+        return $constraintErrorFieldMap[$fieldName];
+    }
+
+    private function getTooLargeConstraintErrorCode(string $fieldName): int
+    {
+        $constraintErrorFieldMap = [
+            'apiClientId' => ApiAccessConstraintException::CLIENT_ID_TOO_LARGE,
+            'clientName' => ApiAccessConstraintException::CLIENT_NAME_TOO_LARGE,
+            'description' => ApiAccessConstraintException::DESCRIPTION_TOO_LARGE,
+        ];
+
+        if (!array_key_exists($fieldName, $constraintErrorFieldMap)) {
+            throw new \RuntimeException(sprintf('"%s" is not mapped with constraint error code', $fieldName));
+        }
+
+        return $constraintErrorFieldMap[$fieldName];
+    }
+
+    private function generateMaxLengthValue(array $data, string $fieldName): array
+    {
+        $length = match ($fieldName) {
+            'apiClientId', 'clientName' => 260,
+            'description' => 21900,
+            default => throw new \RuntimeException(sprintf('The field %s cannot have a max value generated', $fieldName)),
+        };
+        $characters = '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
+        $randomString = '';
+
+        for ($i = 0; $i < $length; ++$i) {
+            $index = rand(0, strlen($characters) - 1);
+            $randomString .= $characters[$index];
+        }
+
+        $data[$fieldName] = $randomString;
+
+        return $data;
```

Comment:
I searched if PHP would have a native function to do this, seems like `random_bytes` should do the job

```suggestion
        return random_bytes($length);
```

Reply:
this is breaking all my test by generating complete giberish like this : 
```
��)�e|�t$�Bz4� �)���
^C�6iƯ :�bF�{��|5r@�����x��Ӭ�Rw�Q\�'��S'J�J���bȔ�栶s�"��`>�
�*�&Vd�4�q)�w��>͛�IA��kr:�S���q7e¢�5�������&�\c{EI��Â(�l6Y��Ҏ�M�}�cL��'�L�
����=,>����)����6���R�[b���wkדּǦb�<4� <#��fۡ0,�/Q@�+��l�la[z��Oi��
```